pkg/crc/api/api.go

```diff
@@ -61,15 +49,36 @@ func (api CrcAPIServer) handleClusterOperations() {
 func (api CrcAPIServer) handleRequest(req commandRequest, conn net.Conn) {
 	defer conn.Close()
 	var result string
-	if handler, ok := api.handlers[req.Command]; ok {
-		config, _, err := api.newConfig()
-		if err != nil {
-			result = encodeErrorToJSON(fmt.Sprintf("Failed to initialize new config store: %v", err))
-		} else {
-			result = handler(api.client, config, req.Args)
-		}
-	} else {
-		result = encodeErrorToJSON(fmt.Sprintf("Unknown command supplied: %s", req.Command))
+
+	config, _, err := api.newConfig()
+	if err != nil {
+		logging.Error(err.Error())
+		result = encodeErrorToJSON(fmt.Sprintf("Failed to initialize new config store: %v", err))
+		writeStringToSocket(conn, result)
+		return
+	}
+
+	switch req.Command {
+	case "start":
+		result = api.handler.Start(config, req.Args)
+	case "stop":
+		result = api.handler.Stop()
+	case "status":
+		result = api.handler.Status()
+	case "delete":
+		result = api.handler.Delete()
+	case "version":
+		result = api.handler.GetVersion()
+	case "setconfig":
+		result = api.handler.SetConfig(config, req.Args)
+	case "unsetconfig":
+		result = api.handler.UnsetConfig(config, req.Args)
+	case "getconfig":
+		result = api.handler.GetConfig(config, req.Args)
+	case "webconsoleurl":
+		result = api.handler.GetWebconsoleInfo()
+	default:
+		result = encodeErrorToJSON("Unexpected error occurred")
```

Comment:
You can keep `result = encodeErrorToJSON(fmt.Sprintf("Unknown command supplied: %s", req.Command))`